### ISSUE TITLE
[BUG] `format-comment` is listing the macros from dart doc

### ISSUE THREAD

**Issue by ValentinVignal** [dart-code-checker/dart-code-metrics#812]
* **Dart code metrics version:**
4.14.0

* **Dart sdk version:**
```
Dart SDK version: 2.16.2 (stable) (Tue Mar 22 13:15:13 2022 +0100) on "macos_x64"
```

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
include: package:flutter_lints/flutter.yaml

analyzer:
  plugins:
    - dart_code_metrics
  strong-mode:
    implicit-casts: false
    implicit-dynamic: true
  errors:
    undefined_prefixed_name: ignore
linter:
  rules:
    always_put_required_named_parameters_first: true
    always_use_package_imports: true
    avoid_bool_literals_in_conditional_expressions: true
    avoid_classes_with_only_static_members: true
    avoid_double_and_int_checks: true
    avoid_dynamic_calls: true
    avoid_escaping_inner_quotes: true
    avoid_field_initializers_in_const_classes: true
    avoid_js_rounded_ints: true
    avoid_multiple_declarations_per_line: true
    avoid_null_checks_in_equality_operators: true
    avoid_private_typedef_functions: true
    avoid_redundant_argument_values: true
    avoid_returning_null_for_future: true
    avoid_setters_without_getters: true
    avoid_shadowing_type_parameters: true
    avoid_slow_async_io: true
    avoid_type_to_string: true
    avoid_types_on_closure_parameters: true
    avoid_unused_constructor_parameters: true
    avoid_void_async: true
    cancel_subscriptions: true
    cascade_invocations: true
    cast_nullable_to_non_nullable: true
    close_sinks: true
    comment_references: true
    conditional_uri_does_not_exist: true
    depend_on_referenced_packages: true
    directives_ordering: true
    empty_constructor_bodies: true
    eol_at_end_of_file: true
    flutter_style_todos: true
    invariant_booleans: true
    join_return_with_assignment: true
    leading_newlines_in_multiline_strings: true
    lines_longer_than_80_chars: false # We use 160: true
    literal_only_boolean_expressions: true
    missing_whitespace_between_adjacent_strings: true
    no_adjacent_strings_in_list: true
    no_logic_in_create_state: true
    no_runtimeType_toString: true
    noop_primitive_operations: true
    null_check_on_nullable_type_parameter: true
    omit_local_variable_types: true
    one_member_abstracts: false # Flutter disabled it because it is useful to have abstract classes with only one member
    only_throw_errors: true
    package_api_docs: true
    parameter_assignments: true
    prefer_asserts_in_initializer_lists: true
    prefer_constructors_over_static_methods: true
    prefer_final_in_for_each: true
    prefer_final_locals: true
    prefer_foreach: true
    prefer_if_elements_to_conditional_expressions: true
    prefer_null_aware_method_calls: true
    prefer_null_aware_operators: true
    prefer_mixin: false # Has false positives, see https://github.com/dart-lang/linter/issues/3018.
    prefer_single_quotes: true
    public_member_api_docs: true
    sized_box_shrink_expand: true
    sort_child_properties_last: true
    sort_constructors_first: true
    sort_pub_dependencies: true
    sort_unnamed_constructors_first: true
    test_types_in_equals: true
    throw_in_finally: true
    tighten_type_of_initializing_formals: true
    unawaited_futures: true
    unnecessary_await_in_return: true
    unnecessary_constructor_name: true
    unnecessary_lambdas: true
    unnecessary_late: true
    unnecessary_null_aware_assignments: true
    unnecessary_null_checks: true
    unnecessary_nullable_for_final_variable_declarations: true
    unnecessary_parenthesis: true
    unnecessary_raw_strings: true
    unnecessary_statements: true
    unsafe_html: true
    use_decorated_box: true
    use_if_null_to_convert_nulls_to_bools: true
    use_is_even_rather_than_modulo: true
    use_key_in_widget_constructors: false
    use_late_for_private_fields_and_variables: true
    use_named_constants: true
    use_raw_strings: true
    use_setters_to_change_properties: true
    use_test_throws_matchers: true
    valid_regexps: true
    void_checks: true

dart_code_metrics:
  anti-patterns:
    - long-method
    - long-parameter-list
  metrics:
    cyclomatic-complexity: 20
    maximum-nesting-level: 5
    number-of-parameters: 4
    source-lines-of-code: 50
  metrics-exclude:
    - "lib/generated_plugin_registrant.dart" # This file is automatically created when you debug on web
    - "**/*.g.dart"
    - "**/*.gql.dart"
  rules:
    - avoid-nested-conditional-expressions
    - no-boolean-literal-compare
    - no-empty-block:
        exclude:
          - test/** # we often pass empty callbacks in tests when they are required
    - prefer-trailing-comma:
        severity: style # default is warning and that would fail on Bitrise
    - no-equal-then-else
    - avoid-border-all
    - avoid-returning-widgets
    # It is a rule name.
    # cSpell:disable-next-line
    - avoid-unnecessary-setstate
    - avoid-wrapping-in-padding
    - prefer-const-border-radius
    - avoid-missing-enum-constant-in-map
    - format-comment
    - no-magic-number
    - prefer-async-await
    - prefer-commenting-analyzer-ignores
    - prefer-first
    - prefer-last

```

</details>

**What did you do? Please include the source code example causing the issue.**

```dart
/// {@template my_template}
/// My widget.
/// {@endtemplate}
class MyWidget extends StatelessWidget {
  /// {@macro my_template}                  <- There is a lint here 
  const MyWidget();
  
  @override
  Widget build(BuildContext context) {
    return const SizedBox();
  }
}
```

The lint:
```
Prefer formatting comments like sentences.
/// {@macro my_template}. dart[format-comment](https://dartcodemetrics.dev/docs/rules/common/format-comment)
```

**What did you expect to happen?**
I would expect no lint to be there.

**What actually happened?**
There is a lint.

**Are you willing to submit a pull request to fix this bug?**

Why not, but I have not looked at the code.

**Comment by incendial** [comment]
@ValentinVignal could you please provide the docs where `@macro` is described?

**Comment by ValentinVignal** [comment]
> @ValentinVignal could you please provide the docs where `@macro` is described?

It comes from dartdoc: https://pub.dev/packages/dartdoc

**Comment by incendial** [comment]
Available in 4.15.0 🚀
